File: PYTHON/Komputer.py
from Cpu import Cpu
from Ram import Ram
from Harddrive import Harddrive
from PowerSupply import PowerSupply
from Gpu import Gpu
from Motherboard import Motherboard
from Case import Case

class Komputer:
    def __init__(self, nama, cpu, ram_list, harddrive, power_supply, gpu, motherboard, casing):
        self.nama = nama
        self.cpu = cpu
        self.ram_list = ram_list
        self.harddrive = harddrive
        self.power_supply = power_supply
        self.gpu = gpu
        self.motherboard = motherboard
        self.casing = casing

    def set_nama(self, nama):
        self.nama = nama

    def set_cpu(self, cpu):
        self.cpu = cpu

    def set_ram(self, ram_list):
        self.ram_list = ram_list

    def set_harddrive(self, harddrive):
        self.harddrive = harddrive

    def set_power_supply(self, power_supply):
        self.power_supply = power_supply

    def set_gpu(self, gpu):
        self.gpu = gpu

    def set_motherboard(self, motherboard):
        self.motherboard = motherboard

    def set_casing(self, casing):
        self.casing = casing

    def add_ram(self, ram):
        self.ram_list.append(ram)

    def get_nama(self):
        return self.nama

    def get_cpu(self):
        return self.cpu

    def get_ram_list(self):
        return self.ram_list

    def get_harddrive(self):
        return self.harddrive

    def get_power_supply(self):
        return self.power_supply

    def get_gpu(self):
        return self.gpu

    def get_motherboard(self):
        return self.motherboard

    def get_casing(self):
        return self.casing

File: CPP/Komputer.cpp
// Komputer.cpp (modifikasi)
#pragma once
#include <iostream>
#include <string>
#include <vector>
#include "Cpu.cpp"
#include "Harddrive.cpp"
#include "Ram.cpp"
#include "PowerSupply.cpp"
#include "Gpu.cpp"
#include "Motherboard.cpp"
#include "Case.cpp"

using namespace std;

class Komputer
{
private:
    string nama;
    Cpu cpu;
    vector<Ram> ramList;
    Harddrive harddrive;
    PowerSupply powerSupply;
    Gpu gpu;
    Motherboard motherboard;
    Case casing;
public:
    Komputer() {}

    Komputer(string nama, Cpu cpu, vector<Ram> ramList, Harddrive harddrive, PowerSupply powerSupply, Gpu gpu, Motherboard motherboard, Case casing)
    {
        this->nama = nama;
        this->cpu = cpu;
        this->ramList = ramList;
        this->harddrive = harddrive;
        this->powerSupply = powerSupply;
        this->gpu = gpu;
        this->motherboard = motherboard;
        this->casing = casing;
    }

    void setNama(string nama)
    {
        this->nama = nama;
    }

    void setCpu(Cpu cpu)
    {
        this->cpu = cpu;
    }

    void setRam(vector<Ram> ramList)
    {
        this->ramList = ramList;
    }

    void setHarddrive(Harddrive harddrive)
    {
        this->harddrive = harddrive;
    }

    void setPowerSupply(PowerSupply powerSupply)
    {
        this->powerSupply = powerSupply;
    }

    void setGpu(Gpu gpu)
    {
        this->gpu = gpu;
    }

    void setMotherboard(Motherboard motherboard)
    {
        this->motherboard = motherboard;
    }

    void setCasing(Case casing)
    {
        this->casing = casing;
    }

    void addRam(Ram ram)
    {
        this->ramList.push_back(ram);
    }

    string getNama()
    {
        return this->nama;
    }

    Cpu getCpu()
    {
        return this->cpu;
    }

    vector<Ram> getRamList()
    {
        return this->ramList;
    }

    Harddrive getHarddrive()
    {
        return this->harddrive;
    }

    PowerSupply getPowerSupply()
    {
        return this->powerSupply;
    }

    Gpu getGpu()
    {
        return this->gpu;
    }

    Motherboard getMotherboard()
    {
        return this->motherboard;
    }

    Case getCasing()
    {
        return this->casing;
    }

    ~Komputer() {}
};
